core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelFile.java

```diff
@@ -467,4 +478,23 @@ public CamelModel getCamelModel(){
 		return CamelModelFactory.getModelForProject(resource != null ? resource.getProject() : null);
 	}
 
+	@Override
+	public List<AbstractCamelModelElement> findAllNodesWithId(String nodeId) {
+		List<AbstractCamelModelElement> result = new ArrayList<>();
+
+		if (getGlobalDefinitions() != null && !getGlobalDefinitions().isEmpty()) {
+			for (AbstractCamelModelElement e : getGlobalDefinitions().values()) {
+				if (e.getId() != null && e.getId().equals(nodeId)) {
```

Comment:
to silmplify code, you can check one time at the beginning of the method if nodeId is !=null and then nodeId.equals(e.getId()) only